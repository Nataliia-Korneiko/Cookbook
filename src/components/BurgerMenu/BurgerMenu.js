import React from 'react';
// import { Link, NavLink } from 'react-router-dom';
import { NavLink } from 'react-router-dom';
import routes from '../../routes/routes';

import s from './BurgerMenu.module.css';

const BurgerMenu = () => {
  return (
    <div className={s.menu__wrap}>
      {/* <div> */}
      <input type="checkbox" className={s.toggle} />
      <div className={s.hamburger}>
        <div />
      </div>
      <div className={s.menu}>
        <div>
          <div>
            <ul>
              <li>
                <NavLink exact to={routes.HOME}>
                  Home
                </NavLink>
              </li>
              <li>
                <NavLink to={routes.ABOUT}>About</NavLink>
              </li>
              <li>
                <NavLink to={routes.MY_RECIPES}>My Recipes</NavLink>
              </li>
              <li>
                <NavLink to={routes.LOG_IN}>Log In</NavLink>
              </li>
              <li>
                <NavLink to={routes.SIGN_UP}>Sign Up</NavLink>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default BurgerMenu;
