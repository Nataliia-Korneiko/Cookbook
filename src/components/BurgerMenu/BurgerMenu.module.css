.toggle {
  position: absolute;
  top: 0;
  left: 0;
  width: 50px;
  height: 50px;
  cursor: pointer;
  opacity: 0;
  z-index: 2;
}

.toggle:checked .hamburger__icon {
  transform: rotate(135deg);
}

.toggle:checked .hamburger__icon::before,
.toggle:checked .hamburger__icon::after {
  top: 0;
  transform: rotate(90deg);
}

.toggle:checked:hover .hamburger__icon {
  transform: rotate(225deg);
}

.toggle:checked ~ .menu {
  visibility: visible;
}

.toggle:checked ~ .menu > div {
  transform: scale(1);
  transition-duration: 0.75s;
}

.hamburger {
  padding: 10px;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50px;
  height: 50px;
  background-color: var(--bg-color-black);
  box-shadow: var(--box-shadow);
  z-index: 1;
}

.hamburger__icon {
  position: relative;
  display: flex;
  flex: none;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 2px;
  background-color: var(--bg-color-golden);
  transition: 0.4s ease-in-out;
}

.hamburger__icon::before,
.hamburger__icon::after {
  position: absolute;
  content: '';
  width: 100%;
  height: 2px;
  background: inherit;
  z-index: 1;
}

.hamburger__icon::before {
  top: -10px;
}

.hamburger__icon::after {
  top: 10px;
}

.menu {
  position: fixed;
  justify-content: center;
  align-items: center;
  display: flex;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  visibility: hidden;
  overflow: hidden;
}

.menu > div {
  display: flex;
  flex: none;
  justify-content: center;
  align-items: center;
  width: 200vw;
  height: 200vw;
  border-radius: 50%;
  background-color: var(--bg-color-golden);
  transform: scale(0);
  transition: 0.4s ease-in-out;
}

.menu > div > div {
  position: relative;
  width: 600px;
  height: 600px;
  background-image: url(../../assets/scroll.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  text-align: center;
}

.list {
  position: absolute;
  top: 50%;
  left: 42%;
  transform: translate(-50%, -50%);
  width: 130px;
}

.item {
  list-style: none;
  color: var(--color-black);
}

.item:not(:last-child) {
  margin-bottom: 20px;
}

.link {
  display: block;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  font-weight: 600;
  color: var(--color-golden);
  text-transform: uppercase;
  text-shadow: var(--text-shadow);
  transition: 0.3s ease-in-out;
}

.link:hover,
.link:focus {
  color: var(--color-white);
  transition: 0.3s ease-in-out;
}

@media (max-width: 660px) {
  .menu > div > div {
    width: 420px;
    height: 420px;
  }

  .list {
    top: 56%;
    width: 140px;
  }

  .item:not(:last-child) {
    margin-bottom: 30px;
  }
}
