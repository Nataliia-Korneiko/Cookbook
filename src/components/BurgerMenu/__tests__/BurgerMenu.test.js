import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import BurgerMenu from '../BurgerMenu';

describe('BurgerMenu component', () => {
  const newBurgerMenu = shallow(<BurgerMenu />);

  it('renders a properly', () => {
    expect(newBurgerMenu).toMatchSnapshot();
  });

  it('renders one input', () => {
    expect(newBurgerMenu.find('input[type="checkbox"]')).toHaveLength(1);
  });

  it('renders five div', () => {
    expect(newBurgerMenu.find('div')).toHaveLength(5);
  });

  it('renders one ul', () => {
    expect(newBurgerMenu.find('ul')).toHaveLength(1);
  });

  it('renders five li', () => {
    expect(newBurgerMenu.find('li')).toHaveLength(5);
  });

  it('renders five NavLink', () => {
    expect(newBurgerMenu.find(NavLink)).toHaveLength(5);
  });
});
