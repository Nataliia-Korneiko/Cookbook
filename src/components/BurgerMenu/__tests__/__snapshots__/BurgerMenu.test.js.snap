// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu component renders a properly 1`] = `ShallowWrapper {}`;
